ansibullbot/utils/extractors.py

```diff
@@ -493,77 +494,63 @@ def metadata(self):
-        inphase = False
+            documentation = u''
+            inphase = False
+            lines = to_text(self.filedata).split(u'\n')
+            for line in lines:
+                if u'DOCUMENTATION' in line:
+                    inphase = True
+                    continue
+                if inphase and (line.strip().endswith((u"'''", u'"""'))):
+                    #phase = None
+                    break
+                if inphase:
+                    documentation += line + u'\n'
+        
+            self._DOCUMENTATION_RAW = documentation
 
-        lines = to_text(self.filedata).split(u'\n')
-        for line in lines:
-            if u'DOCUMENTATION' in line:
-                inphase = True
-                continue
-            if inphase and (line.strip().endswith((u"'''", u'"""'))):
-                #phase = None
-                break
-            if inphase:
-                documentation += line + u'\n'
+            try:
+                self._DOCSTRING = yaml.load(self._DOCUMENTATION_RAW)
+            except:
+                self._DOCSTRING = {}
 
-        if not documentation:
-            logging.debug(u'no documentation found in {}'.format(self.filepath))
-            return []
+            # always cast to a dict for easier handling later
+            if self._DOCSTRING is None:
+                self._DOCSTRING = {}
 
-        # clean out any other yaml besides author to save time
-        inphase = False
-        author_lines = u''
-        self._DOCUMENTATION_RAW = documentation
-        doc_lines = documentation.split(u'\n')
-        for idx, x in enumerate(doc_lines):
-            if x.startswith(u'author'):
-                #print("START ON %s" % x)
-                inphase = True
-                #continue
-            if inphase and not x.strip().startswith((u'-', u'author')):
-                #print("BREAK ON %s" % x)
-                inphase = False
-                break
-            if inphase:
-                author_lines += x + u'\n'
-
-        if not author_lines:
-            logging.debug(u'no author lines found in {}'.format(self.filepath))
-            return []
+        return self._DOCSTRING
 
-        self._AUTHORS_RAW = author_lines
-        ydata = {}
-        try:
-            ydata = yaml.load(author_lines, BotYAMLLoader)
-            self._AUTHORS_DATA = ydata
-        except Exception as e:
-            print(e)
-            return []
+    def get_module_authors(self):
+        """Grep the authors out of the module docstrings"""
 
-        # quit early if the yaml was not valid
-        if not ydata:
-            return []
+        # 2019-02-15
+        if 'author' or 'authors' in self.docs:
+            _authors = self.docs.get('author') or self.docs.get('authors')
+            if _authors is None:
+                return []
+            if not isinstance(_authors, list):
+                _authors = [_authors]
+            logins = set()
+            for author in _authors:
```

Comment:
Now I see that you want just
```suggestion
            return list(set(self.extract_github_id(author) for author in _authors))
```

Reply:
TypeError: unhashable type: 'list'

Reply:
oh why?

Reply:
oh it should probably be
```suggestion
            return list(set(a for author in _authors for a in self.extract_github_id(author)))
```